### PR TITLE
3505 - Add delete data source end point and delete message topics 

### PR DESCRIPTION
**Pull request by minotogna**
https://github.com/openforis/fra-platform/assets/1138156/97ab9437-dfd7-44c5-baa4-a56a55a35582

### CODE REVIEW

**Comment by sorja** [comment]
should the navigation have green bullet when there is a new message? in the video, it is still orange

**Comment by minotogna** [comment]
> should the navigation have green bullet when there is a new message? in the video, it is still orange

Orange is unread and it has higher priority

File: src/client/store/data/slice.ts [hunk 1]
```diff
@@ -121,6 +123,21 @@ export const dataSlice = createSlice({
       Objects.setInPath({ obj: state, path, value })
     })
 
+    builder.addCase(deleteDataSource.pending, (state, action) => {
+      const { assessmentName, cycleName, countryIso, sectionName, uuid } = action.meta.arg
+
+      const name = CommentableDescriptionName.dataSources
+      const value = state.descriptions[assessmentName]?.[cycleName]?.[countryIso]?.[sectionName]?.[name]
+      if (!value) {
+        throw new Error(`Unable to find data source value ${assessmentName}-${cycleName}-${countryIso}-${sectionName}}`)
```
Comment (sorja) [inline src/client/store/data/slice.ts:132]: error message missing `name` param

File: src/client/store/data/slice.ts [hunk 2]
```diff
@@ -121,6 +123,21 @@ export const dataSlice = createSlice({
       Objects.setInPath({ obj: state, path, value })
     })
 
+    builder.addCase(deleteDataSource.pending, (state, action) => {
```
Comment (sorja) [inline src/client/store/data/slice.ts:126]: should we move it under reducer ? or will data slice be refactored later?

**Review by sorja — APPROVED** [review]

**Pull request merged**